input_setup.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# shellcheck disable=SC2086,SC2310,SC2312
+
+# SC2310: This function is invoked in an 'if' condition so set -e will be disabled.
+#         Invoke separately if failures should cause the script to exit.
+#
+# SC2312: Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore).
+
+set -euo pipefail
+
+inputs() {
+  # quoted here because of non-legal names (e.g. check-mode - the "-" gets parsed as subtraction)
+  jq -r ".[\"$1\"] // empty" <<<${SETUP_INPUTS}
+}
+
+payload() {
+  jq -r ".inputs.payload | fromjson | .$1 // empty" ${GITHUB_EVENT_PATH}
+}
+
+githubEventPR() {
+  jq -r ".$1 // empty" <<<${SETUP_GITHUB_EVENT_PR}
+}
+
+# This is different from the other scripts because the INPUT_DEBUG variable doesn't exist yet
+if [[ $(inputs debug) == "true" || $(payload debug) == "true" ]]; then
+  set -x
+fi
+
+# this seems like a no-op?
+cat >>${GITHUB_ENV} <<EOF
+GITHUB_TOKEN=${SETUP_GITHUB_TOKEN}
+EOF
```

Comment:
This is just writing `${{ github.token }}` to the `GITHUB_TOKEN` env var - unsure why it's needed, but I've copied it over just in case